include/coreir/ir/fwd_declare.h

```diff
@@ -141,16 +139,17 @@ typedef std::pair<Wireable*,Wireable*> Connection;
 typedef std::vector<std::pair<Wireable*,Wireable*>> LocalConnections;
 
 
-//Comparison classes for caches
-class ConnectionComp {
+//Comparison classes
+
+//This will do fast comparisons on Connections, but makes no guarentees on being the same between different runs of the program
+class ConnectionCompFast {
   public:
-    static bool SPComp(const SelectPath& l, const SelectPath& r);
     bool operator() (const Connection& l, const Connection& r) const;
 };
 
-class ConnectionStrComp {
+//This will do slow comparisons on Connections but makes guarentee on being consistent between different runs
+class ConnectionCompConsistent {
   public:
-    static bool SPComp(const SelectPath& l, const SelectPath& r);
     bool operator() (const Connection& l, const Connection& r) const;
 };
```

Comment:
Interesting idea. I think it makes sense to have two modes, one for testing/debugging, one for performance. We may want to do a design review and think about how we can fit his pattern into the overall system.

Reply:
This was not quite the intended use case but would be interesting. I intend to have all Connection comparisons be "fast" in the core part of the IR but then when serializing (to verilog or CoreIR or to anything), using the slow deterministic one. The deterministic one is also available if a pass needs to use connections in a map or set deterministically.